packages/block-library/src/list/edit.js

```diff
@@ -13,40 +20,83 @@ export default function ListEdit( {
 	className,
 } ) {
-	const { ordered, values } = attributes;
+	const { ordered, values, reversed, start, type } = attributes;
 
 	return (
-		<RichText
-			identifier="values"
-			multiline="li"
-			tagName={ ordered ? 'ol' : 'ul' }
-			onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
-			value={ values }
-			wrapperClassName="block-library-list"
-			className={ className }
-			placeholder={ __( 'Write list…' ) }
-			onMerge={ mergeBlocks }
-			unstableOnSplit={
-				insertBlocksAfter ?
-					( before, after, ...blocks ) => {
-						if ( ! blocks.length ) {
-							blocks.push( createBlock( 'core/paragraph' ) );
-						}
+		<Fragment>
+			<RichText
+				identifier="values"
+				multiline="li"
+				tagName={ ordered ? 'ol' : 'ul' }
+				onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
+				value={ values }
+				start={ start }
+				reversed={ reversed }
+				type={ type }
+				wrapperClassName="block-library-list"
+				className={ ordered ? classnames( className, 'ol-type-is-' + type ) : className }
+				placeholder={ __( 'Write list…' ) }
+				onMerge={ mergeBlocks }
+				unstableOnSplit={
+					insertBlocksAfter ?
+						( before, after, ...blocks ) => {
+							if ( ! blocks.length ) {
+								blocks.push( createBlock( 'core/paragraph' ) );
+							}
 
-						if ( after !== '<li></li>' ) {
-							blocks.push( createBlock( 'core/list', {
-								ordered,
-								values: after,
-							} ) );
-						}
+							if ( after !== '<li></li>' ) {
+								blocks.push( createBlock( 'core/list', {
+									ordered,
+									values: after,
+								} ) );
+							}
 
-						setAttributes( { values: before } );
-						insertBlocksAfter( blocks );
-					} :
-					undefined
+							setAttributes( { values: before } );
+							insertBlocksAfter( blocks );
+						} :
+						undefined
+				}
+				onRemove={ () => onReplace( [] ) }
+				onTagNameChange={ ( tag ) => setAttributes( { ordered: tag === 'ol' } ) }
+			/>
+			{
+				ordered &&
+				<InspectorControls>
+					<PanelBody title={ __( 'Ordered List Settings' ) }>
+						<SelectControl
+							label={ __( 'List Type' ) }
+							value={ type }
+							options={ [
+								{ label: 'Decimal', value: '1' },
+								{ label: 'Lower alpha', value: 'a' },
+								{ label: 'Upper alpha', value: 'A' },
+								{ label: 'Lower alpha', value: 'i' },
+								{ label: 'Upper alpha', value: 'I' },
+							] }
+							onChange={ ( nextType ) => {
+								setAttributes( { type: nextType } );
+							} }
+						/>
+						<BaseControl label={ __( 'Start Value' ) } >
+							<input
+								type="number"
+								onChange={ ( event ) => {
+									setAttributes( { start: parseInt( event.target.value, 10 ) } );
+								} }
+								value={ start }
+								step="1"
+							/>
+						</BaseControl>
+						<ToggleControl
+							label={ __( 'Reverse List' ) }
+							checked={ !! reversed }
+							onChange={ ( ) => {
+								setAttributes( { reversed: ! reversed } );
```

Comment:
Similar to https://github.com/WordPress/gutenberg/pull/15113/files#r278706296, would it be possible to unset the attribute if it is `false`?